### PR TITLE
Nutrient deficiency module

### LINKED ISSUE #49: Implement a nutrient deficiency module

### PR DESCRIPTION
**Pull request by mirceatlx**
Implement the nutrient deficiency module using the masks generated by DeepLabv3+.

Closes #49

### CODE REVIEW

**Comment by paulmis** [comment]
Let's write tests for the different pipeline modules rather than checking if it works in `main.py`. This way we can tell if something breaks going forward :)

File: src/backend/pipeline/modules/index/index.py [hunk 1]
```diff
@@ -10,4 +11,4 @@ class Index(ParallelModule):
 
     def __init__(self, data: Data, input_data: Any):
         """Initializes the index module and its runnables."""
-        super().__init__(data, [NDVI], name="Index", module_type=Modules.INDEX)
+        super().__init__(data, [Nutrient], name="Index", module_type=Modules.INDEX)
```
Comment (paulmis) [inline src/backend/pipeline/modules/index/index.py:14]: Can you parametrize this in `config.py`?

File: src/backend/pipeline/modules/index/runnables/nutrient.py [hunk 2]
```diff
@@ -18,19 +18,21 @@ def __init__(self, data: Data) -> None:
...
+            result = self.calculate(masks, patches, hsize)
+            data.modules[Modules.INDEX]["runnables"][self.type]["index"] = result
             return True
         except Exception as exception:
             print("Nutrient calculation failed!")
             print(exception)
             return False
 
-    def calculate(self, masks, patches) -> list[np.ndarray]:
+    def calculate(self, masks, patches, hsize: int) -> list[np.ndarray]:
```
Comment (paulmis) [inline src/backend/pipeline/modules/index/runnables/nutrient.py:35]: Missing arg comment for `hsize`

File: src/backend/pipeline/modules/index/index.py (hunk 1 above)
Comment (mirceatlx) [reply src/backend/pipeline/modules/index/index.py]: I made a workaround for this. The only caveat I can see at the moment is that Parallel Modules can only be done before or after Modules. This should be fine as Parallel Modules are used for indicies.

Comment (paulmis) [reply src/backend/pipeline/modules/index/index.py]: This looks good, I would just move the `self.parallel_modules` into the dictionary for module initialization. The module type is `dict[Module, any]` and you can attach the `runnables` to the `any`. In any case, it's alright with me, you can resolve!! :D

**Comment by paulmis** [comment]
LGTM

**Review by paulmis — APPROVED** [review]
LGTM

**Pull request merged**